pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -1311,9 +1488,26 @@ func (nc *Controller) cancelPodEviction(node *v1.Node) bool {
 
 // evictPods queues an eviction for the provided node name, and returns false if the node is already
 // queued for eviction.
-func (nc *Controller) evictPods(node *v1.Node) bool {
+// If node was already evicted, the pods are deleted immediately.
+func (nc *Controller) evictPods(node *v1.Node, pods []*v1.Pod) bool {
 	nc.evictorLock.Lock()
 	defer nc.evictorLock.Unlock()
+	status, ok := nc.nodeEvictionMap.getStatus(node.Name)
+	if ok && status == evicted {
+		// Node eviction already happened for this node.
+		// Handling immediate pod deletion.
+		_, err := nodeutil.DeletePods(nc.kubeClient, pods, nc.recorder, node.Name, string(node.UID), nc.daemonSetStore)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("unable to delete pods from node %q: %v", node.Name, err))
```

Comment:
I think we should requeue in this case, right?

Reply:
Error propagation added. When error is returned to processPod method, pod is requeued.